### PR TITLE
Dịch "backprop_vn" - Phần 2

### LINKED ISSUE #965: Dịch "backprop_vn" - Phần 2
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_multilayer-perceptrons/backprop_vn.md#L39-L114

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by lyphilong**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #965

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 1]
```diff
@@ -42,23 +42,25 @@ To start, we focus our exposition on a three layer (one hidden) multilayer perce
...
 <!--
 Forward propagation refers to the calculation and storage of intermediate variables (including outputs) for the neural network in order from the input layer to the output layer.
 We now work step-by-step through the mechanics of a deep network with one hidden layer.
 This may seem tedious but in the eternal words of funk virtuoso James Brown, you must "pay the cost to be the boss".
 -->
 
-*dịch đoạn phía trên*
-
+Lan truyền ngược đề cập đến việc tính toán cũng như lưu trữ các biến trung gian (bao gồm cả đầu ra) cho mạng nơ-ron theo thứ tự từ tầng đầu vào đến tầng đầu ra.
+Bây giờ ta hãy làm từng bước thông qua cơ chế của mạng nơ-ron sâu với một tầng ẩn.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/backprop_vn.md]:
```suggestion
Bây giờ ta sẽ thực hiện từng bước trong cơ chế làm việc của mạng nơ-ron sâu có một tầng ẩn.
```

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 2]
```diff
@@ -42,23 +42,25 @@ To start, we focus our exposition on a three layer (one hidden) multilayer perce
...
+Điều này có vẻ tẻ nhạt nhưng theo những ca từ bất hủ của nghệ sĩ dòng nhạc funk tài danh James Brown, bạn phải "trả giá để được làm sếp".
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/backprop_vn.md]: câu quote có nên để nguyên tiếng anh không nhỉ, vì nó có vần

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 3]
```diff
@@ -42,23 +42,25 @@ To start, we focus our exposition on a three layer (one hidden) multilayer perce
...
 
 <!--
 For the sake of simplicity, let’s assume that the input example is $\mathbf{x}\in \mathbb{R}^d$ and that our hidden layer does not include a bias term.
 Here the intermediate variable is:
 -->
 
-*dịch đoạn phía trên*
+Để đơn giản, ta giả sử mẫu đầu vào là $\mathbf{x}\in \mathbb{R}^d$ và tầng ẩn của ta sẽ không có hệ số điều chỉnh.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/backprop_vn.md]:
```suggestion
Để đơn giản hóa vấn đề, ta giả sử mẫu đầu vào là $\mathbf{x}\in \mathbb{R}^d$ và tầng ẩn của ta không có hệ số điều chỉnh.
```

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 4]
```diff
@@ -67,7 +69,8 @@ where $\mathbf{W}^{(1)} \in \mathbb{R}^{h \times d}$ is the weight parameter of
 After running the intermediate variable $\mathbf{z}\in \mathbb{R}^h$ through the activation function $\phi$ we obtain our hidden activations vector of length $h$,
 -->
 
-*dịch đoạn phía trên*
+trong đó $\mathbf{W}^{(1)} \in \mathbb{R}^{h \times d}$ là tham số trọng số của tầng ẩn.
+Sau khi chạy biến trung gian $\mathbf{z}\in \mathbb{R}^h$ và cho qua hàm kích hoạt $\phi$, ta nhận được vector kích hoạt ẩn với số chiều $h$,
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/backprop_vn.md]:
```suggestion
Sau khi đưa biến trung gian $\mathbf{z}\in \mathbb{R}^h$ qua hàm kích hoạt $\phi$, ta thu được vector kích hoạt ẩn với $h$ phần tử,
```

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 5]
```diff
@@ -76,23 +79,24 @@ The hidden variable $\mathbf{h}$ is also an intermediate variable.
 Assuming the parameters of the output layer only possess a weight of $\mathbf{W}^{(2)} \in \mathbb{R}^{q \times h}$, we can obtain an output layer variable with a vector length of $q$:
 -->
 
-*dịch đoạn phía trên*
+Biến ẩn $\mathbf{h}$ cũng là một biến trung gian.
+Giả sử các tham số của tầng đầu ra chỉ có trọng số $\mathbf{W}^{(2)} \in \mathbb{R}^{q \times h}$, ta sẽ nhận được một biến tại tầng đầu ra với số chiều vector $q$:
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/backprop_vn.md]: vế sau câu này ở bản gốc lằng nhằng quá, mình xin phép tự sửa lại
```suggestion
Giả sử các tham số của tầng đầu ra chỉ có trọng số $\mathbf{W}^{(2)} \in \mathbb{R}^{q \times h}$, ta sẽ thu được một vector với $q$ phần tử ở tầng đầu ra:
```

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 6]
```diff
@@ -76,23 +79,24 @@ The hidden variable $\mathbf{h}$ is also an intermediate variable.
...
 
 $$\mathbf{o}= \mathbf{W}^{(2)} \mathbf{h}.$$
 
 <!--
 Assuming the loss function is $l$ and the example label is $y$, we can then calculate the loss term for a single data example,
 -->
 
-*dịch đoạn phía trên*
+Giả sử hàm mất mát là $l$ và nhãn của mẫu là $y$, ta có thể tính được lượng mất mát cho mẫu dữ liệu duy nhất,
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/backprop_vn.md]:
```suggestion
Giả sử hàm mất mát là $l$ và nhãn của mẫu là $y$, ta có thể tính được lượng mất mát cho một mẫu dữ liệu duy nhất,
```

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 7]
```diff
@@ -101,15 +105,16 @@ where the Frobenius norm of the matrix is simply the $L_2$ norm applied after fl
 Finally, the model's regularized loss on a given data example is:
 -->
 
-*dịch đoạn phía trên*
+trong đó chuẩn Frobenius của ma trận chỉ đơn giản là lấy chuẩn $L_2$ sau khi trải ma trận thành vector.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/backprop_vn.md]: mình sửa thành `trải phẳng` do đợt trước sót mất commit trong glossary
```suggestion
trong đó chuẩn Frobenius của ma trận chỉ đơn giản là chuẩn $L_2$ của vector thu được sau khi trải phẳng ma trận.
```

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 8]
```diff
@@ -101,15 +105,16 @@ where the Frobenius norm of the matrix is simply the $L_2$ norm applied after fl
...
+Cuối cùng, hàm mất mát điều chuẩn của mô hình trên một mẫu dữ liệu cho trước là:
 
 $$J = L + s.$$
 
 <!--
 We refer to $J$ the *objective function* in the following discussion.
 -->
 
-*dịch đoạn phía trên*
+Ta sẽ bàn thêm về *hàm mục tiêu* $J$ trong các phần sau này.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/backprop_vn.md]:
```suggestion
Ta sẽ bàn thêm về *hàm mục tiêu* $J$ ở phía dưới.
```

File: chapter_multilayer-perceptrons/backprop_vn.md [hunk 9]
```diff
@@ -42,23 +42,25 @@ To start, we focus our exposition on a three layer (one hidden) multilayer perce
 ## Forward Propagation
 -->
 
-## *dịch tiêu đề phía trên*
+## Lan truyền Ngược
```
[the remark this answers is not in the document]
Comment (lyphilong) [reply chapter_multilayer-perceptrons/backprop_vn.md]: @minhduc0711 chu choa tai hại quá :)))

File: chapter_multilayer-perceptrons/backprop_vn.md (hunk 2 above)
Comment (lyphilong) [reply chapter_multilayer-perceptrons/backprop_vn.md]: @minhduc0711 Phần này em cũng nghĩ giống anh, do câu tiếng anh nghe hay hơn câu tiếng việt.

Comment (minhduc0711) [reply chapter_multilayer-perceptrons/backprop_vn.md]: mình không chắc 2 câu này có nghĩa giống nhau không :v

Comment (duythanhvn) [reply chapter_multilayer-perceptrons/backprop_vn.md]: Mình thấy cũng ổn, vẫn là step-by-step. Mặc dù mình khá thích từ **trả giá**, kể cả khi để nguyên hay dịch ra thì mình thấy nó cũng hơi kỳ (mình không chắc nhiều người biết James Brown của những năm 60-70 đâu).

Comment (minhduc0711) [reply chapter_multilayer-perceptrons/backprop_vn.md]: Mình cũng không biết James Brown (chắc phải gọi @duythanhvn bằng anh rồi :sweat_smile:)
Vậy mình đồng ý dịch như @lkhphuc.

Comment (duythanhvn) [reply chapter_multilayer-perceptrons/backprop_vn.md]: > Mình cũng không biết James Brown (chắc phải gọi @duythanhvn bằng anh rồi 😅)
> Vậy mình đồng ý dịch như @lkhphuc.

Yep. I'm old man, bro. 😉

**Comment by duythanhvn** [comment]
@lyphilong Done phần này thôi em ơi.

**Comment by lyphilong** [comment]
> @lyphilong Done phần này thôi em ơi.

em done rồi á anh, mấy nay em bị quên phần này
